tests/regression/test_mesh_from_plex.py

```diff
@@ -0,0 +1,44 @@
+from firedrake import *
+import numpy as np
+
+
+def remove_pyop2_label(plex):
+    plex.removeLabel("pyop2_core")
+    plex.removeLabel("pyop2_owned")
+    plex.removeLabel("pyop2_ghost")
+    return plex
+
+
+def get_plex_with_update_coordinates(mesh):
+    """
+    Update the coordinates of the dmplex in mesh and return a copy of the dmplex
+    """
+    tdim = mesh.topological_dimension()
+    gdim = mesh.geometric_dimension()
+    entity_dofs = np.zeros(tdim + 1, dtype=np.int32)
+    entity_dofs[0] = gdim
+    coord_section = mesh.create_section(entity_dofs)
+    plex = mesh.topology_dm
```

Comment:
Clone plex (and remove pyop2 labels) here?

Reply:
Done!